### PR TITLE
clone selected code on init

### PR DESCRIPTION
**Pull request by moshe-kabala**
- Add return isCreateNew and clone the latest version if existed
- Fix force
- change watch to notWait

### CODE REVIEW

File: cmd/code/push.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 
 func NewPushCmd() *cobra.Command {
 	var secretId string
-	var watch bool
+	var notWait bool
```
Comment (YuvalFeldman) [inline cmd/code/push.go:15]: Can you define as --no-wait at least for how the user sets it?

File: cmd/code/push.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func NewPushCmd() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			if watch && code.IsCodeParsing(currentVersion) {
+			if !notWait && code.IsCodeParsing(currentVersion) {
```
Comment (YuvalFeldman) [inline cmd/code/push.go:49]: Double negative validations are always a bad idea.
I would define a param above:
const watch = !notWait
and give it a comment even to explain why

File: cmd/projects/init.go [hunk 3]
```diff
@@ -21,16 +21,28 @@ func NewInitCmd() *cobra.Command {
...
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := cmd.Context()
-			selectedProject, err := project.GetProjectFromFlag(ctx, projectId, true)
+			selectedProject, _, err := project.GetProjectFromFlag(ctx, projectId, true)
 			if err != nil {
 				return err
 			}
 
-			codeIntegration, err := code.GetCodeIntegrationFromFlag(ctx, codeIntegrationId, true)
+			codeIntegration, isCreatedNewCodeIntegration, err := code.GetCodeIntegrationFromFlag(ctx, codeIntegrationId, true)
```
Comment (YuvalFeldman) [inline cmd/projects/init.go:29]:
```suggestion
			codeIntegration, createdNewCodeIntegration, err := code.GetCodeIntegrationFromFlag(ctx, codeIntegrationId, true)
```

File: pkg/code/utils.go [hunk 4]
```diff
@@ -33,24 +33,24 @@ func CreateCodeIntegration(ctx context.Context, codeIntegrations []CodeIntegrati
 	return AddCodeIntegration(ctx, name)
 }
 
-func GetCodeIntegrationFromFlag(ctx context.Context, codeIntegrationIdFlag string, askForNewProjectFirst bool) (*CodeIntegration, error) {
+func GetCodeIntegrationFromFlag(ctx context.Context, codeIntegrationIdFlag string, askForNewProjectFirst bool) (code *CodeIntegration, isCreatedNew bool, err error) {
```
Comment (YuvalFeldman) [inline pkg/code/utils.go:36]:
```suggestion
func GetCodeIntegrationFromFlag(ctx context.Context, codeIntegrationIdFlag string, askForNewProjectFirst bool) (code *CodeIntegration, createNewCodeIntegration bool, err error) {
```
Best to keep names to make it easy to follow

**Review by YuvalFeldman — APPROVED** [review]
LGTM, see comments

File: cmd/projects/init.go (hunk 3 above)
Comment (moshe-kabala) [reply cmd/projects/init.go]: Changed to `wasCreated`

**Pull request merged**